lib/ansible/galaxy/collection.py

```diff
@@ -1384,7 +1384,19 @@ def _download_file(url, b_path, expected_hash, validate_certs, headers=None):
 
 def _extract_tar_dir(tar, dirname, b_dest):
     """ Extracts a directory from a collection tar. """
-    tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+    try:
+        tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+    except KeyError:
+        # Some archives contain directories that end with '/'
+        if not dirname.endswith(os.path.sep):
+            dirname += os.path.sep
```

Comment:
Currently, you just retry the same thing again if there's a `/` already. Maybe you should drop it?
```suggestion
        # Some archives contain directories that end with '/'
        if not dirname.endswith(os.path.sep):
            dirname += os.path.sep
        else:
            dirname = dirname[:-1]
```

Reply:
Good point. Most of the items do not have a trailing slash. There's room for optimization here. I may end up doing a loop as @jborean93 suggested below.